index.html

```diff
@@ -1615,6 +1618,211 @@ <h3>
+      <p>
+        To <dfn>process the `scope_extensions` member</dfn>, given
+        [=ordered map=] |json:ordered map| and [=ordered map=]
+        |manifest:ordered map|:
+      </p>
+      <ol class="algorithm">
+        <li>Let |processedScopeExtensions:list| be a new [=list=].
+        </li>
+        <li>Set |manifest|["scope_extensions"] to |processedScopeExtensions|.
+        </li>
+        <li>If |json|["scope_extensions"] doesn't [=map/exist=] or is not a 
+          [=list=], return.
+        </li>
+        <li>[=list/For each=] |entry:ordered map| of |json|["scope_extensions"]:
+          <ol>
+            <li>If |entry|["type"] or |entry|["origin"] is undefined,
+              [=iteration/continue=].
+            </li>
+            <li>If |entry|["type"] is not "origin", [=iteration/continue=].
+            </li>
+            <li>If |entry|["origin"] is not a [=string=],
+              [=iteration/continue=].
+            </li>
+            <li>Set |entry|["origin"] to the result of parsing it as an
+              [=origin=].
+            </li>
+            <li>Let [=URL=] |resolved_scope:URL| be an empty [=URL=]. </li>
+            <li>Validate |entry| using the algorithm from [=validate scope
+              extensions=], given [=ordered map=] |entry:ordered map|, [=URL=]
+              |manifest_id:URL|, and [=URL=] |resolved_scope:URL|. 
+            </li>
+            <li>If validation failed, [=iteration/continue=].
+            </li>
+            <li>
+              Set |entry|["scope"] to |resolved_scope|.
+            </li>
+            <li>[=list/Append=] |entry| to |processedScopeExtensions|.
+            </li>
+          </ol>
+        </li>
+      </ol>
+      <aside class="note">
+        `origin` is the only type of scope extension entry that is currently
+        specified. Other types may be added in the future to support different
+        use cases. 
+      </aside>
+      </section>
+      <section class="informative">
+        <h3>The web-app-origin-association file</h3>
+        <p>A <dfn>web-app-origin-association</dfn> file is a JSON file that can
+          be used to [=validate scope extensions=]. It confirm an association
+          between the origin it is in with one or more web applications. It 
+          identifies apps uniquely by referencing manifest [=manifest/ids=]. 
+        </p>
+        <p>
+          Given [=origin=] |origin:origin|, a [=web-app-origin-association=]
+          file is expected to be downloadable from 
+          <code>[origin]/.well-known/web-app-origin-association</code>.
+        </p>
+      </section>
+      <section>
+      <h2>Validating scope extensions</h2>
+      <p>
+        To <dfn>validate scope extensions</dfn>, given [=ordered map=] 
+        |extension:ordered map|, [=URL=] |manifest_id:URL| and [=URL=] 
+        |resolved_scope:URL|:
+      </p>
+      <ol class="algorithm">
+        <li>Set |resolved_scope| to empty URL.</li>
```

Comment:
is it normal to accept an argument like this as an 'out' argument? Do other algorithms in the manifest spec do this?

Reply:
Addressed by not doing this anymore.